src/Diagrams/TwoD/Image.hs

```diff
@@ -94,15 +94,15 @@ instance Fractional n => HasOrigin (DImage n a) where
 image :: (TypeableFloat n, Typeable a, Renderable (DImage n a) b)
       => DImage n a -> QDiagram b V2 n Any
 image img
-  = mkQD (Prim img)
+  = transform tr
+  $ mkQD (Prim img)
```

Comment:
Wouldn't this make `img` get transformed twice? I think you want an `img'` with the transform removed here.

Reply:
Hmm, I'm not sure. I guess it depends how individual backend `Renderable DImage` instances work?   I also just realized that the transformation probably needs to be used in the `HasQuery` instance as well.  This needs some more thought & reworking before it is ready to merge.

Reply:
The backend can't tell the difference between a transform applied before or after the construction of the diagram. At render time all transforms get applied to the `Prim` before the backend sees it. But in this case `img` already has a `tr` so at render time the backend will see `tr <> tr` (plus any other transforms applied to the diagram).

You're right about the `HasQuery` case. I guess I commented out the `transform t` to be consistent with this function but what I did doesn't really make sense. We're also missing `Enveloped` and `Traced` instances. If we write out the proper versions of those instances then we could write

```
image = mkQD (Prim img) (getEnvelope img) (getTrace img) mempty (getQuery img)
```

Reply:
I guess it's worth noting that in my rewrite I don't transform the `Prim`s before giving it to the backend. Instead with each `Prim` you get a `Transform` and `Style` attached. This way we [don't need these placeholder transforms](https://github.com/cchalmers/diagrams/blob/master/src/Diagrams/TwoD/Image.hs#L90) in the primitives like `Image` or `Text`. I feel like this leads to an overall improvement and avoids confusion for things like this issue but it's one of the many changes we need to start discussing!